docker-images/kafka/Dockerfile

```diff
@@ -21,10 +21,15 @@ ENV KAFKA_VERSION=${KAFKA_VERSION}
 ENV STRIMZI_VERSION=${strimzi_version}
 ENV SCALA_VERSION=2.12
 ENV JMX_EXPORTER_VERSION=0.1.0
+ENV LOG4_LOGSTASH_JSON_LAYOUT_VERSION=1.7
 
 # Set Kafka (SHA512) and Prometheus JMX exporter (SHA1) checksums
 ENV KAFKA_CHECKSUM="${KAFKA_SHA512}  kafka_${SCALA_VERSION}-${KAFKA_VERSION}.tgz"
 ENV JMX_EXPORTER_CHECKSUM="6ab370edccc2eeb3985f4c95769c26c090d0e052 jmx_prometheus_javaagent-${JMX_EXPORTER_VERSION}.jar"
+ENV LOG4_LOGSTASH_JSON_LAYOUT_CHECKSUM="507713504f0ddb75ba512f62763519c43cf46fde jsonevent-layout-${LOG4_LOGSTASH_JSON_LAYOUT_VERSION}.jar"
+ENV JSON_SMART_CHECKSUM="24a2f903d25e004de30ac602c5b47f2d4e420a59 json-smart-1.1.1.jar"
```

Comment:
Can we have the versions here configurable as well? You added ENV_VAR for the JSON Layout, you should probably do it for these two as well. It might be also handy to add an explanatory comment that commons-lang and json-smart are beign downloaded because they are dependencies of the JSON Layout.

Reply:
Ok, i think this is very bad behave to specify version of maven dependecies explictly.And adding two more boilerplate `CHECKSUM` AND `VERSION` to dockerfile is aslo not good.
So i rewrtite this solution to usong `pom.xml` and `mvn dependency:copy-dependencies` with help of docker multy stage build. 

@scholzj @stanlyDoge @tombentley Please review it again.
I hope this idea is much better.